servicelayer/logs.py

```diff
@@ -1,12 +1,75 @@
 import logging
+import sys
+
+import structlog
+from structlog.contextvars import merge_contextvars
 
 from servicelayer import settings
 
+LOG_FORMAT_TEXT = "TEXT"
+LOG_FORMAT_JSON = "JSON"
+
+
+def configure_logging(level=logging.INFO):
+    """Configure log levels and structured logging"""
+    common_processors = [
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.add_logger_name,
+        structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f"),
+        structlog.processors.StackInfoRenderer(),
+        structlog.dev.set_exc_info,
+        structlog.processors.format_exc_info,
+        structlog.processors.UnicodeDecoder(),
+    ]
+    if settings.LOG_FORMAT == LOG_FORMAT_TEXT:
+        processors = common_processors
+        formatter = structlog.stdlib.ProcessorFormatter(
+            foreign_pre_chain=processors, processor=structlog.dev.ConsoleRenderer(),
+        )
+    else:
+        processors = common_processors + [
+            merge_contextvars,
+            format_stackdriver,
+        ]
+        formatter = structlog.stdlib.ProcessorFormatter(
+            foreign_pre_chain=processors, processor=structlog.processors.JSONRenderer(),
+        )
+
+    # configuration for structlog based loggers
+    structlog.configure(
+        processors=processors
+        + [structlog.stdlib.ProcessorFormatter.wrap_for_formatter,],
+        logger_factory=structlog.stdlib.LoggerFactory(),
+    )
+
+    # handler for low level logs that should be sent to STDOUT
```

Comment:
I'm not 100% sure this does the correct thing in terms of filtering out the `level` given to the function... and maybe it wants to be called `out_handler`? I am still not seeing `DEBUG` messages on my local dev instance.

Reply:
`DEBUG` messages don't show because the default level is `INFO` https://github.com/alephdata/servicelayer/pull/46/files#diff-a51f18d073c472478dbcc6331c34971cb246cc27f69035a7cc240868e9a95168R13. I have some new changes in Aleph and Memorious that set the level according to the `DEBUG` setting.

Reply:
+1 to `out_handler`